src/NumericalAlgorithms/Interpolation/InterpolationTargetReceiveVars.hpp

```diff
@@ -201,28 +214,98 @@ void callback_and_cleanup(
           completed_ids->pop_front();
         }
       });
+}
 
-  // Tell interpolators to clean up at this temporal_id for this
-  // InterpolationTargetTag.
-  auto& interpolator_proxy =
-      Parallel::get_parallel_component<Interpolator<Metavariables>>(*cache);
-  Parallel::simple_action<Actions::CleanUpInterpolator<InterpolationTargetTag>>(
-      interpolator_proxy, temporal_id);
-
-  // If we have a sequential target, and there are further
-  // temporal_ids, begin interpolation for the next one.
-  if (InterpolationTargetTag::compute_target_points::is_sequential::value) {
-    const auto& temporal_ids = db::get<Tags::TemporalIds<TemporalId>>(*box);
-    if (not temporal_ids.empty()) {
-      auto& my_proxy = Parallel::get_parallel_component<
-          InterpolationTarget<Metavariables, InterpolationTargetTag>>(*cache);
-      Parallel::simple_action<
-          typename InterpolationTargetTag::compute_target_points>(
-          my_proxy, temporal_ids.front());
+/// Returns true if this InterpolationTarget has received data
+/// at all its points.
+///
+/// have_data_at_all_points is called by an Action of InterpolationTarget.
+///
+/// Currently two Actions call have_data_at_all_points:
+///   InterpolationTargetReceiveVars (called by Interpolator ParallelComponent)
+///   InterpolationTargetVarsFromElement (called by DgElementArray)
+template <typename InterpolationTargetTag, typename DbTags, typename TemporalId>
+bool have_data_at_all_points(const db::DataBox<DbTags>& box,
+                             const TemporalId& temporal_id) noexcept {
+  const size_t filled_size =
+      db::get<Tags::IndicesOfFilledInterpPoints<TemporalId>>(box)
+          .at(temporal_id)
+          .size();
+  const size_t invalid_size = [&box, &temporal_id ]() noexcept {
+    const auto& invalid_indices =
+        db::get<Tags::IndicesOfInvalidInterpPoints<TemporalId>>(box);
+    if (invalid_indices.count(temporal_id) > 0) {
+      return invalid_indices.at(temporal_id).size();
     }
+    return 0_st;
   }
+  ();
+  const size_t interp_size =
+      db::get<Tags::InterpolatedVars<InterpolationTargetTag, TemporalId>>(box)
+          .at(temporal_id)
+          .number_of_grid_points();
+  return (invalid_size + filled_size == interp_size);
 }
 
+/// Adds the supplied interpolated variables and offsets to the
+/// InterpolationTarget's internal DataBox.
+///
+/// add_received_variables is called by an Action of InterpolationTarget.
+///
+/// Currently two Actions call add_received_variables:
+///   InterpolationTargetReceiveVars (called by Interpolator ParallelComponent)
+///   InterpolationTargetVarsFromElement (called by DgElementArray)
+template <typename InterpolationTargetTag, typename DbTags, typename TemporalId>
+void add_received_variables(
+    const gsl::not_null<db::DataBox<DbTags>*> box,
+    const std::vector<Variables<
+        typename InterpolationTargetTag::vars_to_interpolate_to_target>>&
+        vars_src,
```

Comment:
The documentation says "the supplied interpolated variables", which sounds like the _result_ of the interpolation. But this argument is "vars_to_interpolate_to_target", which sounds like _input_ to the interpolation. Can the dox and/or the variable name be clarified?

Reply:
The tag for the input vars is the same as the tag for the output vars, since it is the same set of variables (but at a different set of points).  I will try to clarify that in the documentation.